subsys/bluetooth/host/hci_core.c

```diff
@@ -18,6 +18,7 @@
 #include <zephyr/sys/byteorder.h>
 #include <zephyr/debug/stack.h>
 #include <zephyr/sys/__assert.h>
+#include <soc.h>
```

Comment:
This is a bit surprising. What does hci_core.c need this for? I'd have expected to see reports of build failures, etc if it was absolutely necessary there.